src/Connectors/src/ConnectorBase/Services/UriInfo.cs

```diff
@@ -66,8 +66,8 @@ public UriInfo(string uristring, string username, string password)
                 Query = GetQuery(uri.PathAndQuery);
             }
 
-            UserName = username;
-            Password = password;
+            UserName = WebUtility.UrlEncode(username?.Replace("+", "-plus-")?.Replace(" ", "-space-"));
```

Comment:
@TimHess Originally, the bug appears that we did not have a `UrlEncode` here but we do now. So if there is a `+` or `<space>` in the username or password, it will now be converted to `%2B` and a `+` respectively. And the subsequent `UrlDecode` will revert to the original. So it doesn't look like we need the `Replace("+", "-plus-")?.Replace(" ", "-space-"))`. Am I missing something ?

Reply:
Yeah, you're right, this solution was over-thought